cache/remote.go

```diff
@@ -293,6 +303,11 @@ func (p lazyRefProvider) ReaderAt(ctx context.Context, desc ocispecs.Descriptor)
 
 func (p lazyRefProvider) Unlazy(ctx context.Context) error {
 	_, err := p.ref.cm.unlazyG.Do(ctx, string(p.desc.Digest), func(ctx context.Context) (_ interface{}, rerr error) {
+		defer func() {
+			if rerr == nil {
+				rerr = p.ref.addBlob(ctx, p.desc)
```

Comment:
Should this defer be after the check below for whether the ref is lazy or not?

Reply:
Thank you for the review. Moved the defer to after the check.